pkg/translate/translateValue.go

```diff
@@ -185,130 +161,107 @@ func (t *ValueYAMLTranslator) TranslateFromValueToSpec(values *v1alpha2.Values)
 
 // TranslateTree translates input value.yaml Tree to ControlPlaneSpec Tree
 func (t *ValueYAMLTranslator) TranslateTree(valueTree map[string]interface{}, cpSpecTree map[string]interface{}, path util.Path) error {
-
+	// translate enablement and namespace
+	err := t.setEnablementAndNamespacesFromValue(valueTree, cpSpecTree)
+	if err != nil {
+		return fmt.Errorf("error when translating enablement and namespace from value.yaml tree %v", err.Error())
+	}
 	// translate with api mapping
-	err := translateTree(valueTree, cpSpecTree, path, t.APIMapping)
+	err = translateTree(valueTree, cpSpecTree, t.APIMapping)
 	if err != nil {
 		return fmt.Errorf("error when translating value.yaml tree with global mapping %v", err.Error())
 	}
 	// translate with k8s mapping
-	err = translateTree(valueTree, cpSpecTree, path, t.KubernetesMapping)
+	err = translateTree(valueTree, cpSpecTree, t.KubernetesMapping)
 	if err != nil {
 		return fmt.Errorf("error when translating value.yaml tree with kubernetes mapping %v", err.Error())
 	}
-	// translate enablement and namespace
-	err = t.setEnablementAndNamespacesFromValue(cpSpecTree, valueTree)
-	if err != nil {
-		return fmt.Errorf("error when translating enablement and namespace from value.yaml tree %v", err.Error())
-	}
 	return nil
 }
 
-func firstCharToLowerPath(input string) util.Path {
-	var path util.Path
-	for _, p := range util.PathFromString(input) {
-		p = firstCharToLower(p)
-		path = append(path, p)
-	}
-	return path
-}
-
 // setEnablementAndNamespaces translates the enablement and namespace value of each component in the baseYAML values
 // tree, based on feature/component inheritance relationship.
-func (t *ValueYAMLTranslator) setEnablementAndNamespacesFromValue(root map[string]interface{}, valueSpec map[string]interface{}) error {
-	for vp, fe := range t.FeatureEnablementMapping {
-		enabled := name.IsComponentEnabledFromValue(vp, valueSpec)
-		// set feature enablement
-		if fe == nil || fe.outPath == "" {
-			continue
+func (t *ValueYAMLTranslator) setEnablementAndNamespacesFromValue(valueSpec map[string]interface{}, root map[string]interface{}) error {
+	for cnv, cni := range t.ValuesToComponentName {
+		enabled, err := name.IsComponentEnabledFromValue(cnv, valueSpec)
+		if err != nil {
+			return err
 		}
-		newP := firstCharToLowerPath(fe.outPath)
-		// Value.yaml component to IstioFeature is N:1, so if the feature is enabled by other component already, skip setting
-		curEnabled, found, _ := name.GetFromValuePath(root, newP)
-		if !found || curEnabled == false {
-			if err := tpath.WriteNode(root, newP, enabled); err != nil {
-				return err
-			}
+		featureName := name.ComponentNameToFeatureName[cni]
+		tmpl := componentEnablementPattern
+		if strings.Contains(cnv, "gateway") {
```

Comment:
This is not robust. Instead create a function that tests if it's a gateway based on name.ComponentName.

Reply:
changed to check based on component name